src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -118,6 +118,24 @@ public static string GetFolderPath(SpecialFolder folder, SpecialFolderOption opt
             return GetFolderPathCore(folder, option);
         }
 
+        private static int s_processId;
```

Comment:
Could be a `readonly static` and not use the `volatile`?

Or is it to lazy eval and avoid initing everything when `Environment.xx` is accessed?

Reply:
> Or is it to lazy eval and avoid initing everything when Environment.xx is accessed?

Yes